### PR TITLE
Address several kernel and hyperv-daemons cves

### PR DESCRIPTION
**Pull request by rlmenge**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary 

Address kernel cves

###### Change Log  

- Address CVE-1999-0656, CVE-2007-4998, CVE-2010-0298, CVE-2010-0309, CVE-2010-4563, CVE-2019-3819, CVE-2019-3887, CVE-2020-25672, CVE-2021-20194, CVE-2021-3564, CVE-2021-3669, CVE-2022-38096, CVE-2023-46838, CVE-2023-6531, CVE-2023-6931, CVE-2024-0646, CVE-2024-21803, CVE-2024-23849, CVE-2024-23850, CVE-2024-23851, CVE-2024-25739, CVE-2024-26900, CVE-2024-35984, CVE-2024-35990, CVE-2024-35997, CVE-2024-36008, CVE-2024-0565, CVE-2024-23307, CVE-2024-24857, CVE-2024-24858, CVE-2024-24859, CVE-2024-24860, CVE-2024-24861, CVE-2024-26913, CVE-2024-27022, CVE-2024-35992, CVE-2024-36891, CVE-2024-36893, CVE-2024-36897, CVE-2024-36901, CVE-2024-36902, CVE-2024-36925, CVE-2024-36926, CVE-2024-36930, CVE-2024-36938, CVE-2022-2785, CVE-2023-6200, CVE-2024-26949, CVE-2024-26952, CVE-2024-36971, CVE-2019-10638, CVE-2019-19076, CVE-2019-25160, CVE-2021-45480, CVE-2023-31084, CVE-2023-3338, CVE-2024-26602, CVE-2024-35848, CVE-2024-35849, CVE-2024-35852, CVE-2024-35853, CVE-2024-35854, CVE-2024-35857, CVE-2024-36008, CVE-2024-36009

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
kernel
- https://nvd.nist.gov/vuln/detail/CVE-1999-0656
- https://nvd.nist.gov/vuln/detail/CVE-2007-4998
- https://nvd.nist.gov/vuln/detail/CVE-2010-0298
- https://nvd.nist.gov/vuln/detail/CVE-2010-0309
- https://nvd.nist.gov/vuln/detail/CVE-2010-4563
- https://nvd.nist.gov/vuln/detail/CVE-2019-3819
- https://nvd.nist.gov/vuln/detail/CVE-2019-3887
- https://nvd.nist.gov/vuln/detail/CVE-2020-25672
- https://nvd.nist.gov/vuln/detail/CVE-2021-20194
- https://nvd.nist.gov/vuln/detail/CVE-2021-3564
- https://nvd.nist.gov/vuln/detail/CVE-2021-3669
- https://nvd.nist.gov/vuln/detail/CVE-2022-38096
- https://nvd.nist.gov/vuln/detail/CVE-2023-46838
- https://nvd.nist.gov/vuln/detail/CVE-2023-6531
- https://nvd.nist.gov/vuln/detail/CVE-2023-6931
- https://nvd.nist.gov/vuln/detail/CVE-2024-0646
- https://nvd.nist.gov/vuln/detail/CVE-2024-21803
- https://nvd.nist.gov/vuln/detail/CVE-2024-23849
- https://nvd.nist.gov/vuln/detail/CVE-2024-23850
- https://nvd.nist.gov/vuln/detail/CVE-2024-23851
- https://nvd.nist.gov/vuln/detail/CVE-2024-25739
- https://nvd.nist.gov/vuln/detail/CVE-2024-26900
- https://nvd.nist.gov/vuln/detail/CVE-2024-35984
- https://nvd.nist.gov/vuln/detail/CVE-2024-35990
- https://nvd.nist.gov/vuln/detail/CVE-2024-35997
- https://nvd.nist.gov/vuln/detail/CVE-2024-36008
- https://nvd.nist.gov/vuln/detail/CVE-2024-0565
- https://nvd.nist.gov/vuln/detail/CVE-2024-23307
- https://nvd.nist.gov/vuln/detail/CVE-2024-24857
- https://nvd.nist.gov/vuln/detail/CVE-2024-24858
- https://nvd.nist.gov/vuln/detail/CVE-2024-24859
- https://nvd.nist.gov/vuln/detail/CVE-2024-24860
- https://nvd.nist.gov/vuln/detail/CVE-2024-24861
- https://nvd.nist.gov/vuln/detail/CVE-2024-26913
- https://nvd.nist.gov/vuln/detail/CVE-2024-27022
- https://nvd.nist.gov/vuln/detail/CVE-2024-35992
- https://nvd.nist.gov/vuln/detail/CVE-2024-36891
- https://nvd.nist.gov/vuln/detail/CVE-2024-36893
- https://nvd.nist.gov/vuln/detail/CVE-2024-36897
- https://nvd.nist.gov/vuln/detail/CVE-2024-36901
- https://nvd.nist.gov/vuln/detail/CVE-2024-36902
- https://nvd.nist.gov/vuln/detail/CVE-2024-36925
- https://nvd.nist.gov/vuln/detail/CVE-2024-36926
- https://nvd.nist.gov/vuln/detail/CVE-2024-36930
- https://nvd.nist.gov/vuln/detail/CVE-2024-36938
- https://nvd.nist.gov/vuln/detail/CVE-2022-2785
- https://nvd.nist.gov/vuln/detail/CVE-2023-6200
- https://nvd.nist.gov/vuln/detail/CVE-2024-26949
- https://nvd.nist.gov/vuln/detail/CVE-2024-26952
- https://nvd.nist.gov/vuln/detail/CVE-2024-36971

hyperv-daemons
- https://nvd.nist.gov/vuln/detail/CVE-2019-10638
- https://nvd.nist.gov/vuln/detail/CVE-2019-19076
- https://nvd.nist.gov/vuln/detail/CVE-2019-25160
- https://nvd.nist.gov/vuln/detail/CVE-2021-45480
- https://nvd.nist.gov/vuln/detail/CVE-2023-31084
- https://nvd.nist.gov/vuln/detail/CVE-2023-3338
- https://nvd.nist.gov/vuln/detail/CVE-2024-26602
- https://nvd.nist.gov/vuln/detail/CVE-2024-35848
- https://nvd.nist.gov/vuln/detail/CVE-2024-35849
- https://nvd.nist.gov/vuln/detail/CVE-2024-35852
- https://nvd.nist.gov/vuln/detail/CVE-2024-35853
- https://nvd.nist.gov/vuln/detail/CVE-2024-35854
- https://nvd.nist.gov/vuln/detail/CVE-2024-35857
- https://nvd.nist.gov/vuln/detail/CVE-2024-36008
- https://nvd.nist.gov/vuln/detail/CVE-2024-36009

Note remaining kernel cves:
```
CVE-2022-4543	no fix upstream
CVE-2024-23848	4 patches not backported yet (see https://ubuntu.com/security/CVE-2024-23848)
CVE-2024-24864	no fix upstream
CVE-2024-25740	no fix upstream
CVE-2024-26596	not backported
CVE-2024-27012	not backported
CVE-2024-27017	not backported
```
Note remaining hyperv-daemons cves:
```
CVE-2024-27397	not backported
CVE-2024-27396	not backported
```

### CODE REVIEW

File: SPECS/kernel/CVE-1999-0656.nopatch [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CVE-1999-0656 impacts a daemon called rpc.ugidd.  This daemon is not provided by Mariner
```
Comment (christopherco) [inline SPECS/kernel/CVE-1999-0656.nopatch:1]: nit - "Mariner" -> "Azure Linux"

File: SPECS/kernel/CVE-2007-4998.nopatch [hunk 2]
```diff
@@ -0,0 +1,7 @@
+This nopatch clears CVE-2007-4998.  The CVE is misrepresentd on NIST as impacting the kernel.  It actually impacts coreutils and
+busybox, but neither of those packages are impacted by this CVE either.
+
+Exploit instructions are to run the cp utility with certain arguments, see https://bugzilla.redhat.com/show_bug.cgi?id=356471 for details
+
+The result on Mariner is that cp reports warnings and the exploit is not effective.
```
Comment (christopherco) [inline SPECS/kernel/CVE-2007-4998.nopatch:6]: nit - "Mariner" -> "Azure Linux"

File: SPECS/kernel/CVE-2010-4563.nopatch [hunk 3]
```diff
@@ -0,0 +1 @@
+CVE-2010-4563 - No upstream fix. Low security impact. No plans to fix.
```
Comment (gjswalling) [inline SPECS/kernel/CVE-2010-4563.nopatch:1]: I usually see comments for nopatch files explaining why it's already fixed or not applicable, but this seems different.  Is nopatch correct for this case?  Wouldn't we want the CVE scanner to still call out if we are vulnerable on this?

**Review by gjswalling — APPROVED** [review]

File: SPECS/kernel/CVE-2024-21803.nopatch [hunk 4]
```diff
@@ -0,0 +1,3 @@
+CVE-2024-21803 - fixed in 6.6.X
```
Comment (Camelron) [inline SPECS/kernel/CVE-2024-21803.nopatch:1]: I am not able to track down this patch for some reason, the link gives me access denied and have been unable to set up an account. https://bugzilla.openanolis.cn/show_bug.cgi?id=8081

File: SPECS/kernel/CVE-2024-24859.nopatch [hunk 5]
```diff
@@ -0,0 +1,3 @@
+CVE-2024-24859 - in version 6.6.26.1 (note that this CVE is not mentioned in the description but is fixed by this patch)
+upstream: 7835fcfd132eb88b87e8eb901f88436f63ab60f7
```
Comment (Camelron) [inline SPECS/kernel/CVE-2024-24859.nopatch:2]: I see this as the fixing patch for this one? https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.35&id=f56e715ef1c19c42c6aa6cb9280947dea13aab2e

Comment (Camelron) [reply SPECS/kernel/CVE-2024-24859.nopatch:2]: nevermind, got this mixed up with CVE-2024-24860

**Review by Camelron — APPROVED** [review]
Double-checked and lgtm :)

File: SPECS/kernel/CVE-2024-21803.nopatch (hunk 4 above)
Comment (rlmenge) [reply SPECS/kernel/CVE-2024-21803.nopatch:1]: I found it through https://ubuntu.com/security/CVE-2024-21803 who determined this was a duplicate of CVE-2023-51779 who has this as a patch https://nvd.nist.gov/vuln/detail/CVE-2023-51779

File: SPECS/kernel/CVE-2010-4563.nopatch (hunk 3 above)
Comment (rlmenge) [reply SPECS/kernel/CVE-2010-4563.nopatch]: Agreed. I have adjusted the wording

**Review by christopherco — APPROVED** [review]

**Pull request merged**